lib/spack/spack/environment.py

```diff
@@ -820,6 +820,15 @@ def config_scopes(self):
         """A list of all configuration scopes for this environment."""
         return self.included_config_scopes() + [self.env_file_config_scope()]
 
+    def set_config(self, path, value):
+        """Set configuration for this environment"""
+        yaml = config_dict(self.yaml)
+        keys = spack.config.process_config_path(path)
+        for key in keys[:-1]:
+            yaml = yaml[key]
+        yaml[keys[-1]] = value
+        self.write()
+
```

Comment:
I'm wondering whether this can bee done through `spack.config` -- should environments subclass `SingleFileScope` and construct a new class that updates the env when the config is edited?